src/main/java/me/botsko/prism/actions/BlockAction.java

```diff
@@ -30,574 +30,534 @@
+        switch (state.getType()) {
+            case SPAWNER:
+                final SpawnerActionData spawnerActionData = new SpawnerActionData();
+                final CreatureSpawner spawner = (CreatureSpawner) state;
+                spawnerActionData.entityType = spawner.getSpawnedType().name().toLowerCase();
+                spawnerActionData.delay = spawner.getDelay();
+                actionData = spawnerActionData;
+                break;
+            case PLAYER_WALL_HEAD:
+            case PLAYER_HEAD:
+                SkullActionData headActionData = new SkullActionData();
+                if (state instanceof Skull) {
+                    Skull skull = ((Skull) state);
+                    if (skull.getOwningPlayer() != null) {
+                        headActionData.owner = skull.getOwningPlayer().getUniqueId().toString();
+                    }
+                }
+                setBlockRotation(state, headActionData);
+                actionData = headActionData;
+                break;
+            case SKELETON_SKULL:
+            case SKELETON_WALL_SKULL:
+            case WITHER_SKELETON_SKULL:
+            case WITHER_SKELETON_WALL_SKULL:
+                SkullActionData skullActionData = new SkullActionData();
+                setBlockRotation(state, skullActionData);
+                actionData = skullActionData;
+                break;
+            case COMMAND_BLOCK:
+                final CommandBlock cmdblock = (CommandBlock) state;
+                final CommandActionData commandActionData = new CommandActionData();
+                commandActionData.command = cmdblock.getCommand();
+                actionData = commandActionData;
+                break;
+            default:
+                if (Tag.SIGNS.isTagged(state.getType())) {
+                    final SignActionData signActionData = new SignActionData();
+                    final Sign sign = (Sign) state;
+                    signActionData.lines = sign.getLines();
+                    actionData = signActionData;
+                }
+        }
+    }
+
+    private void setBlockRotation(BlockState block, SkullActionData skullActionData) {
+        if (block.getBlockData() instanceof Rotatable) {
+            final Rotatable r = (Rotatable) block.getBlockData();
+            skullActionData.rotation = r.getRotation().toString();
+        } else {
+            final Directional d = (Directional) block.getBlockData();
+            skullActionData.rotation = d.getFacing().name().toLowerCase();
+        }
+    }
+
+
+    @Override
+    public boolean hasExtraData() {
+        return actionData != null;
+    }
+
+    @Override
+    public String serialize() {
+        return gson().toJson(actionData);
+    }
+
+    @Override
+    public void deserialize(String data) {
+        if (data != null && data.startsWith("{")) {
+            if (getMaterial() == Material.PLAYER_HEAD || getMaterial() == Material.PLAYER_WALL_HEAD) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unnecessary use of fully qualified name 'Material.PLAYER_WALL_HEAD' due to existing static import 'org.bukkit.Material.*'](https://app.codacy.com/gh/AddstarMC/Prism-Bukkit/pullRequest?prid=5446892)